### PR TITLE
[FIX] product: enable uniqueness of barcode even on archived products

### PR DESCRIPTION
**Pull request by mamatch**
### Steps
- Create a product with a barcode.
- Archive it.
- Try to create another product with the same barcode.

### Issue
You are able to do that but barcodes are meant to be unique per product.

### Reason
The domain used to check for other products with the same barcode doesn't consider the archived ones

opw-3477371

### CODE REVIEW

File: addons/product/models/product_product.py [hunk 1]
```diff
@@ -175,7 +175,7 @@ def _check_barcode_unique(self):
         # Collect `product.product.id` of objects that share a barcode with any of those in `self`,
         # and group them by barcode.
         candidate_duplicates = self.env['product.product'].sudo().read_group(
-            domain=[("barcode", "in", list(set(product.barcode for product in self if product.barcode)))],
+            domain=[("barcode", "in", list(set(product.barcode for product in self if product.barcode))), ("active", "=?", False)],
```
Comment (nle-odoo) [inline addons/product/models/product_product.py:178]: I don't understand the use of =? the ? should work if the value is False or NULL, but the = will work with False, so how are active=True records gotten?

if the intention is to only get both active and non-active product, you could try to add `.with_context(active_test=False)`  before the read group.

**Review by nle-odoo — APPROVED** [review]
looks good to me if it's what we want functionally

File: addons/product/models/product_product.py (hunk 1 above)
Comment (mamatch) [reply addons/product/models/product_product.py:178]: Yes it's exactly what I want to do, thanks!

**Comment by Whenrow** [comment]
The actual behavior is the wanted one (see comment on ticket)

**Pull request closed without merging**